File: BotBuilderNodeJs/libraries/botframework-schema/src/sharepoint/primaryTextCardParameters.ts
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { BaseCardParameters } from './baseCardParameters';
import { ICardParameters } from './ICardParameters';

/**
 * Sharepoint primary text card view parameters
 */
export class PrimaryTextCardParameters extends BaseCardParameters implements ICardParameters {
    private primaryText: string;
    private description: string;

    /**
     * Initializes a new instance of the PrimaryTextCardParameters class
     */
    public PrimaryTextCardParameters(){
        // Do nothing
    }

    /**
     * Sets primary text property of type string
     */
    public set PrimaryText(primaryText: string){
        this.primaryText = primaryText;
    }

    /**
     * Gets primary text property of type string
     */
    public get PrimaryText(): string {
        return this.primaryText;
    }

    /**
     * Sets description property of type string
     */
    public set Description(description: string){
        this.description = description;
    }

    /**
     * Gets description property of type string
     */
    public get Description(): string {
        return this.description;
    }
}

File: BotBuilder DotNet/libraries/Microsoft.Bot.Schema/SharePoint/PrimaryTextCardParameters.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

using System;
using System.Collections.Generic;
using System.Text;
using Newtonsoft.Json;

namespace Microsoft.Bot.Schema.SharePoint
{
    /// <summary>
    /// SharePoint primary text card parameters.
    /// </summary>
    public class PrimaryTextCardParameters : BaseCardParameters, ICardParameters
    {
        /// <summary>
        /// Initializes a new instance of the <see cref="PrimaryTextCardParameters"/> class.
        /// </summary>
        public PrimaryTextCardParameters()
        {
            // Do nothing
        }

        /// <summary>
        /// Gets or Sets the primary text of type <see cref="string"/>.
        /// </summary>
        /// <value>This value is the primary text to display.</value>
        [JsonProperty(PropertyName = "primaryText")]
        public string PrimaryText { get; set; }

        /// <summary>
        /// Gets or Sets the description of type <see cref="string"/>.
        /// </summary>
        /// <value>This value is the description of the card view.</value>
        [JsonProperty(PropertyName = "description")]
        public string Description { get; set; }
    }
}
